mesonbuild/linkers/detect.py

```diff
@@ -92,7 +93,10 @@ def guess_win_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
         elif not invoked_directly:
             return ClangClDynamicLinker(
                 for_machine, override, exelist=compiler, prefix=comp_class.LINKER_PREFIX,
-                version=search_version(o), direct=False, machine=None)
+                version=search_version(o), direct=False, machine=None,
+                rsp_syntax=RSPFileSyntax.MSVC
+                            if (invoked_directly or comp_class.get_argument_syntax() == 'msvc')
```

Comment:
That code is inside the block `elif not invoked_directly:` so the condition can be simplified to `if comp_class.get_argument_syntax() == 'msvc'` AFAIK.

Reply:
Actually, I don't even know what `invoked_directly` really means here? I'm not expert in that part of the code.